torax/geometry.py

```diff
@@ -879,6 +880,197 @@ def from_fbt(
         hires_fac=hires_fac,
     )
 
+  @classmethod
+  def from_eqdsk(
+        cls,
+        geometry_dir: str | None = None,
+        geometry_file: str = 'eqdsk_cocos02.eqdsk',
+        hires_fac: int = 4,
+        Ip_from_parameters: bool = True,
+        n_rho: int = 25,
+    ):
+        """NOTE: Different from other geometry loaders, the shaping factors are coming from eqdsk"""
+
+        
+        def calculate_area(x, z):
+            """Gauss-shoelace formula (https://en.wikipedia.org/wiki/Shoelace_formula)"""
+            n = len(x)
+            area = 0.0
+            for i in range(n):
+                j = (i + 1) % n  # roll over at n
+                area += x[i] * z[j]
+                area -= z[i] * x[j]
+            area = abs(area) / 2.0
+            return area
+
+        eqfile = geometry_loader.load_geo_data(geometry_dir, geometry_file, geometry_loader.GeometrySource.EQDSK)
+
+        Rmaj = eqfile['xmag']
+        Rmin = eqfile['xbdry'].max() - Rmaj
+        B0   = eqfile['bcentre']
+
+        psi_eqdsk_1dgrid = np.linspace(eqfile['psimag'], eqfile['psibdry'], eqfile['nx'])
+
+        X_1D = np.linspace(eqfile['xgrid1'], eqfile['xgrid1'] + eqfile['xdim'], eqfile['nx'])
+        Z_1D = np.linspace(eqfile['zmid'] - eqfile['zdim']/2, eqfile['zmid'] + eqfile['zdim']/2, eqfile['nz'])
+        X, Z = np.meshgrid(X_1D, Z_1D, indexing='ij')
+        Xlcfs, Zlcfs = eqfile['xbdry'], eqfile['zbdry']
+
+        # Psi 2D grid defined on the Meshgrid
+        psi_eqdsk_2dgrid  = eqfile['psi']
+        # Create mask for the confined region, i.e.,Xlcfs.min() < X < Xlcfs.max(), Zlcfs.min() < Z < Zlcfs.max()
+
+        offset = 0.01
+        mask = (X > Xlcfs.min() - offset) & (X < Xlcfs.max() + offset) & (Z > Zlcfs.min() - offset) & (Z < Zlcfs.max() + offset)
+        masked_psi_eqdsk_2dgrid = np.ma.masked_where(~mask, psi_eqdsk_2dgrid)
+
+        # q on uniform grid (pressure, etc., also defined here)
+        q_eqdsk_1dgrid = eqfile['qpsi']
+
+        # ---- Interpolations
+        q_interp = scipy.interpolate.interp1d(psi_eqdsk_1dgrid, eqfile['qpsi'], kind='cubic')
+        psi_spline_fit   = scipy.interpolate.RectBivariateSpline(X_1D, Z_1D, psi_eqdsk_2dgrid, kx=3, ky=3, s=0)
```

Comment:
out of curiosity, did you check robustness of the results to the hyperparameters of this method?

Reply:
IIRC, I did do some testing with this, but determined the resolution of the underlying eqdsk file is more important than the hyperparameters of the spline fitting. Perhaps a warning is of use for this.